caput/pipeline.py

```diff
@@ -565,9 +627,12 @@ def _setup_task(self, task_spec):
                     msg = "Parameter group %s not found in config." % param_key
                     raise PipelineConfigError(msg)
 
+        # add global params to params
+        params.update(self.all_tasks_params)
```

Comment:
I think you want the parameter update to be done the other way around. i.e. local params should be used in preference to global ones.